docs/content/docs/dev/table/concepts/overview.md

```diff
@@ -51,51 +51,470 @@ state is used within a pipeline. The planner decides whether state is necessary
+or through user configuration (see [`table-exec-source-cdc-events-duplicate`]({{< ref "docs/dev/table/config" >}}#table-exec-source-cdc-events-duplicate)).
+
+The following figure illustrates a `SELECT ... FROM` statement that querying an [upsert kafka source]({{< ref "docs/connectors/table/upsert-kafka" >}}).
+```sql
+CREATE TABLE upsert_kakfa (
+    id INT PRIMARY KEY NOT ENFORCED,
+    message  STRING
+) WITH (
+    'connector' = 'upsert-kafka',
+    ...
+);
+
+SELECT * FROM upsert_kakfa;
+```
+The table source only provides messages with *INSERT*, *UPDATE_AFTER* and *DELETE* type, while the downstream sink requires a complete changelog (including *UPDATE_BEFORE*). 
+As a result, although this query itself does not involve explicit stateful calculation, the planner still generates a stateful operator called "ChangelogNormalize" to help obtain the complete changelog.
+{{< img alt="Implicit-derived stateful op" src="/fig/table-streaming/implicit-derived-stateful-op.png" width="85%">}}
+
+{{< hint info >}}
+Please refer to the individual operator documentation for more details about how much state is required
+and how to limit a potentially ever-growing state size.
+{{< /hint >}}
 
 #### Idle State Retention Time
 
 The *Idle State Retention Time* parameter [`table.exec.state.ttl`]({{< ref "docs/dev/table/config" >}}#table-exec-state-ttl)
 defines for how long the state of a key is retained without being updated before it is removed.
-For the previous example query, the count of a`sessionId` would be removed as soon as it has not
+For the previous example query, the count of a`word` would be removed as soon as it has not
 been updated for the configured period of time.
 
 By removing the state of a key, the continuous query completely forgets that it has seen this key
 before. If a record with a key, whose state has been removed before, is processed, the record will
 be treated as if it was the first record with the respective key. For the example above this means
-that the count of a `sessionId` would start again at `0`.
+that the count of a `word` would start again at `0`.
+
+#### Configure Operator-level State TTL
+--------------------------
+{{< hint warning >}}
+This is an advanced feature and should be used with caution. It is only suitable for the cases
+in which there are multiple states used in the pipeline,
+and you need to set different TTL (Time-to-Live) for each state. 
+If the pipeline does not involve stateful computations, you do not need to follow this procedure.
+If the pipeline only uses one state, you only need to set [`table.exec.state.ttl`]({{< ref "docs/dev/table/config" >}}#table-exec-state-ttl)
+at pipeline level.
+{{< /hint >}}
+
+From Flink v1.18, Table API & SQL supports configuring fine-grained state TTL at operator-level to improve the state usage. 
+To be more specific, the number of used states can be defined as the configuration granularity and is associated with each input state of the operator. 
```

Comment:
Due to the one-to-many relationship between the operator and its state, we define the configurable minimum granularity as the number of incoming edges for each state operator. For OneInputStreamOperator, it can configure one state's TTL; for TwoInputStreamOperator (such as Join), which has two inputs, it can configure TTL for the left state and the right state, respectively. More generally, for KInputStreamOperator, K state TTLs can be configured. Assuming that a job uses two stateful OneInputStreamOperators and one stateful TwoInputStreamOperator, the total number of states that can be configured is 1 * 2 + 2 = 4.